components/support/jwcrypto/src/direct.rs

```diff
@@ -0,0 +1,227 @@
+    // prefer consumers to use serde with the entire key and the only use-case
+    // is currently tests, so let's limit this to tests for now...
+    #[cfg(test)]
+    pub fn new_direct_from_bytes(kid: Option<String>, key: &[u8]) -> Self {
+        Jwk {
+            kid,
+            key_parameters: JwkKeyParameters::Direct {
+                k: base64::encode_config(&key, base64::URL_SAFE_NO_PAD),
+            },
+        }
+    }
+}
+
+pub(crate) fn encrypt_to_jwe(
+    data: &[u8],
+    enc: EncryptionAlgorithm,
+    jwk: &Jwk,
+) -> Result<CompactJwe> {
+    // It's slightly unfortunate we need to supply a struct with ECDH specific
+    // values all specified as None, but doesn't seem likely to ever actually hurt.
+    let protected_header = JweHeader {
+        kid: jwk.kid.clone(),
+        alg: Algorithm::Direct,
+        enc,
+        epk: None,
+        apu: None,
+        apv: None,
+    };
+    let secret = match &jwk.key_parameters {
+        JwkKeyParameters::Direct { k } => base64::decode_config(&k, base64::URL_SAFE_NO_PAD)?,
+        _ => return Err(JwCryptoError::IllegalState("Not a Direct key")),
+    };
+    aes::aes_gcm_encrypt(data, protected_header, &secret)
+}
+
+pub(crate) fn decrypt_jwe(jwe: &CompactJwe, jwk: Jwk) -> Result<String> {
+    let secret = match jwk.key_parameters {
+        JwkKeyParameters::Direct { k } => base64::decode_config(&k, base64::URL_SAFE_NO_PAD)?,
+        _ => return Err(JwCryptoError::IllegalState("Not a Direct key")),
+    };
+    aes::aes_gcm_decrypt(jwe, &secret)
+}
+
+#[test]
+fn test_simple_roundtrip() {
+    // We should be able to round-trip data.
+    use super::{decrypt_jwe, encrypt_to_jwe, DecryptionParameters, EncryptionParameters};
+    let jwk = Jwk::new_direct_key(Some("my key".to_string())).unwrap();
+    let data = "to be, or not 🐝🐝";
+    let encrypted = encrypt_to_jwe(
+        data.as_bytes(),
+        EncryptionParameters::Direct {
+            jwk: &jwk,
+            enc: EncryptionAlgorithm::A256GCM,
+        },
+    )
+    .unwrap();
+    let decrypted = decrypt_jwe(&encrypted, DecryptionParameters::Direct { jwk }).unwrap();
+    assert_eq!(data, decrypted);
+}
+
+#[test]
+fn test_modified_ciphertext() {
+    // Modifying the ciphertext will fail.
+    use super::{decrypt_jwe, encrypt_to_jwe, DecryptionParameters, EncryptionParameters};
+    let jwk = Jwk::new_direct_key(Some("my key".to_string())).unwrap();
+    let data = "to be, or not 🐝🐝";
+    let encrypted = encrypt_to_jwe(
+        data.as_bytes(),
+        EncryptionParameters::Direct {
+            jwk: &jwk,
+            enc: EncryptionAlgorithm::A256GCM,
+        },
+    )
+    .unwrap();
+    // additional text
+    assert!(matches!(
+        decrypt_jwe(
+            &(encrypted.clone() + "A"),
+            DecryptionParameters::Direct { jwk: jwk.clone() }
+        ),
+        Err(JwCryptoError::IllegalState(_))
+    ));
+    // truncated text
```

Comment:
Please also test changing one of the bytes in the text without altering its length (because I expect there are length checks that would fail the add/remove cases you've got here but would allow a modified text through to lower layers of the stack).